docs/docs/development/new-behavior.md

```diff
@@ -0,0 +1,449 @@
+  retro-tap:
+    type: boolean
+  hold-trigger-key-positions:
+    type: array
+    required: false
+    default: []
+```
+
+We see that the `.yaml` files used for new behaviors' devicetree bindings consist of the following properties:
+
+#### `description`
+
+A brief statement of what the behavior is. The value of this property is not seen by end-users; as such, the `description` value should be kept less than a sentence long, leaving explanations for end-users of how the behavior works for its documentation.
+
+#### `compatible`
+
+Allows ZMK to assign the correct driver to the behavior extracted from the keymap or `.dtsi`. The value of the `compatible` property is equal to the name of the [devicetree binding file](#creating-the-devicetree-binding-yaml) as a `string`.
+
+As shown in the example above, `compatible: "zmk,behavior-hold-tap"` is the value of the `compatible` property of `zmk,behavior-hold-tap.yaml`.
+
+#### `include`
+
+Choose between `zero_param.yaml`, `one_param.yaml`, or `two_param.yaml` depending on how many additional parameters are required to complete the behavior's binding in a keymap. For example, we `include: two_param.yaml` in `zmk,behavior-hold-tap.yaml` because any user-defined or pre-defined instances of the hold-tap behavior take in two cells as inputs: one for the hold behavior and one for the tap behavior.
+
+#### `properties` (Optional)
+
+These are additional variables required to configure a particular instance of a behavior. `properties` can be of the following types:
+
+- `path`
+- `compound`
+- `array`
+- `string`
+- `string-array`
+- `boolean`
+- `int`
+- `uint8-array`
+- `phandle`.
+- `phandle-array`
+- `phandles`
+
+:::info
+For more information on additional `properties`, refer to [Zephyr's documentation on Devicetree bindings](https://docs.zephyrproject.org/latest/build/dts/bindings.html#properties).
+:::
+
+### Creating the driver (`.c`)
+
+:::info
+Developing drivers for behaviors in ZMK makes extensive use of the Zephyr Devicetree API and Device Driver Model. Links to the Zephyr Project Documentation for both of these concepts can be found below:
+
+- [Zephyr Devicetree API](https://docs.zephyrproject.org/latest/build/dts/api/api.html)
+- [Zephyr Device Driver Model](https://docs.zephyrproject.org/latest/kernel/drivers/index.html)
+
+:::
+
+Driver files are stored in `app/src/behaviors/` and are labelled in lowercase, beginning with the prefix `behavior_`, and ending with the behavior's name, using underscores to separate multiple words. For example, the directory for the hold-tap's driver would be located at `app/src/behaviors/behavior_hold_tap.c`.
+
+The code snippet below shows the essential components of a new driver.
+
+```c
+#define DT_DRV_COMPAT zmk_<behavior_name>
+
+// Dependencies
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+
+#include <zmk/behavior.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
```

Comment:
Planning to rewrite this section given the review done by @joelspadin in #1275.
https://github.com/zmkfirmware/zmk/pull/1275#discussion_r862518614

Reply:
Leave it for a separate MR maybe? I think it would be good to just get this merged and then follow up.